### DISCUSSION TITLE
Packages/com.unity.inputsystem/api/UnityEngine.InputSystem.InputActionReference

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#687] [Notes]
# Packages/com.unity.inputsystem/api/UnityEngine.InputSystem.InputActionReference

https://docs.unity3d.com/Packages/com.unity.inputsystem@latest/index.html?subfolder=%2Fapi%2FUnityEngine.InputSystem.InputActionReference.html

**Comment by vertxxyz** [comment]
An InputAction is an instance that isn't shared across InputActionMaps, so if you mix and match Input Actions created by script, and the assets, they will refer to different instances and you must resolve one to the other if you intend to have identical bindings across the two.
